bridge/engine/EngineStreamDirector.h

```diff
@@ -873,11 +882,14 @@ class EngineStreamDirector
         // Unpinned, belogns to lastN and some of the participants needs this quality.
         for (const auto& participant : _participantStreams)
         {
-            if (participant.first != senderEndpointIdHash && participant.second.desiredUnpinnedLevel == quality)
+            if (participant.first != senderEndpointIdHash &&
+                quality == std::min(participant.second.desiredUnpinnedLevel, highestAvailableQuality))
```

Comment:
other check we can add if `senderEndpointIdHash` is pinned for this participant. If it's, the unpinned will not be used for this participant

Reply:
Your logic is correct, but as you said: _".. If it's, the unpinned will not be used for THIS participant"_, we can not early break on it, because we need to know whether this quality is needed at all (by at least one participant). So the fact that one participant needs _pinned_ quality from our sender, does not exclude possibility that some other participants need _unpinned_ quality from this very sender.

Reply:
Yes, I know I am just saying could continue iterate because imagine all participants are able to receive the highest quality for dominant speaker, then packets for this layer does not need to be decrypted.